### PR TITLE
Added support for network configuration for windows vm

### PR DESCRIPTION
**Pull request by anandnevase**
Added support for network configuration (DHCP as well as Static ip) for windows vm. 

For windows provision, following are prerequisites:
1. win_username and win_password required in profile
2. win_username is always "Administrator" only in VM template.

### CODE REVIEW

File: salt/cloud/clouds/vmware.py [hunk 1]
```diff
@@ -776,23 +820,22 @@ def _wait_for_ip(vm_ref, max_wait):
...
-
     time_counter = 0
     starttime = time.time()
     while time_counter < max_wait_ip:
         if time_counter % 5 == 0:
             log.info("[ {0} ] Waiting to retrieve IPv4 information [{1} s]".format(vm_ref.name, time_counter))
 
-        if vm_ref.summary.guest.ipAddress and match(IP_RE, vm_ref.summary.guest.ipAddress) and vm_ref.summary.guest.ipAddress != '127.0.0.1':
-            log.info("[ {0} ] Successfully retrieved IPv4 information in {1} seconds".format(vm_ref.name, time_counter))
+	if vm_ref.summary.guest.ipAddress and valid_ip(vm_ref.summary.guest.ipAddress):
```
Comment (ghost) [inline salt/cloud/clouds/vmware.py]: I think the tabs are mixed with spaces here. Can you fix it by using 4 spaces for all the tabs?

Comment (anandnevase) [inline salt/cloud/clouds/vmware.py]: Updated line indentation

File: salt/cloud/clouds/vmware.py [hunk 2]
```diff
@@ -776,21 +819,21 @@ def _wait_for_ip(vm_ref, max_wait):
...
+    
     time_counter = 0
     starttime = time.time()
     while time_counter < max_wait_ip:
         if time_counter % 5 == 0:
             log.info("[ {0} ] Waiting to retrieve IPv4 information [{1} s]".format(vm_ref.name, time_counter))
 
-        if vm_ref.summary.guest.ipAddress and match(IP_RE, vm_ref.summary.guest.ipAddress) and vm_ref.summary.guest.ipAddress != '127.0.0.1':
-            log.info("[ {0} ] Successfully retrieved IPv4 information in {1} seconds".format(vm_ref.name, time_counter))
+    	if vm_ref.summary.guest.ipAddress and _valid_ip(vm_ref.summary.guest.ipAddress):
```
Comment (ghost) [inline salt/cloud/clouds/vmware.py]: Still there are mixed tabs.

Comment (anandnevase) [inline salt/cloud/clouds/vmware.py]: done

**Pull request merged**